### PR TITLE
Truncate access review roles with badge list

### PR DESCRIPTION
**Pull request by lukkor**
## Summary

- Replace GraphQL `AccessEntry.role` with `roles: [String!]!`, split from the stored comma-separated `role` string at the API boundary (no DB or driver changes)
- Add `AccessEntryRoles()` helper and unit tests for empty, single, and multi-role values
- Add `AccessEntryRolesCell` to the campaign detail table: up to 3 neutral badges, then a `+X` badge that opens a popover with remaining roles

## Test plan

- [x] `go test ./pkg/server/api/console/v1/types/... -run AccessEntryRoles`
- [x] Open an access review campaign with a user that has many roles (e.g. Cloudflare) and confirm the Role column shows 3 badges plus a `+X` popover
- [x] Confirm a single-role or empty-role entry still renders correctly (`NotAvailable` when empty)

image

Closes ENG-459.

---
## Summary by cubic
Expose access-review roles as an array end-to-end and render up to three role badges with a +X popover in the console to fix layout issues from long role strings (ENG-459). Adds a DB `roles` array column and updates drivers and services to emit and persist multiple roles.

### Tests **`+259`** **`-144`**
- Update driver/coredata tests to assert `Roles []string`.
- Add HubSpot role-merging tests and an upsert test for nil→empty roles.

### Coredata **`+37`** **`-11`**
- Add `roles TEXT[]` with backfill migration; drop `role`.
- Switch `AccessReviewEntry` to `Roles []string` and default nil to `[]`; update queries and upserts.

### GraphQL API **`+7`** **`-2`**
- Replace `role: String!` with `roles: [String!]!` on `AccessReviewEntry`, defaulting to an empty array.

### MCP **`+12`** **`-5`**
- Update spec and types to expose `roles: string[]`.

### Service **`+361`** **`-213`**
- Replace `AccountRecord.Role` with `Roles []string` and pass arrays through the review engine.
- Update all drivers to populate role arrays and normalize per-provider (e.g., Cloudflare defaults to ["Member"], Microsoft 365 keeps all assigned roles, Datadog collects multiple roles, HubSpot merges role IDs and adds "Super Admin" when applicable).

### App: console **`+92`** **`-3`**
- Add `AccessEntryRolesCell` to show up to 3 neutral badges and a “+X” popover using `@radix-ui/react-popover` and `@probo/ui`; integrate in the campaign table and update queries to use `roles`.

<sup>Written for commit 8094e7cfd03a7ad9d9e76ba530720168dc5b9a0d. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/getprobo/probo/pull/1351?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: pkg/server/api/console/v1/access_review_campaign_resolvers.go [hunk 1]
```diff
@@ -984,3 +984,15 @@ type accessReviewCampaignConnectionResolver struct{ *Resolver }
 type accessReviewCampaignScopeSourceResolver struct{ *Resolver }
 type accessSourceResolver struct{ *Resolver }
 type accessSourceConnectionResolver struct{ *Resolver }
+
+// !!! WARNING !!!
```
Comment (codenem) [inline pkg/server/api/console/v1/access_review_campaign_resolvers.go]: to remove

**Review by codenem — APPROVED** [review]

**Review by gearnode — APPROVED** [review]
Let fix the issue on the database first.

**Comment by lukkor** [comment]
@cubic-dev-ai review this PR again

**Comment by lukkor** [comment]
@cubic-dev-ai review this PR again

**Review by gearnode — APPROVED** [review]
LGTM
